### PR TITLE
Basic database functionality

### PR DESCRIPTION
**Pull request by KiranitSS**

### CODE REVIEW

File: Data/Contexts/ProductsDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (this.Database.GetPendingMigrations().Any())
+            {
+                this.Database.Migrate();
+            }
+        }
+
+        public virtual DbSet<Category> Categories { get => this.Set<Category>(); }
```
Comment (XmyriyCat) [inline Data/Contexts/ProductsDbContext.cs:26]: Why not use just { get; } ?)

File: DomainModels/Customer.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace DomainModels;
+
+public class Customer
+{
+    public int Id { get; set; }
+
+    public string Name { get; set; } = string.Empty;
+
+    public string? PhoneNumber { get; set; }
```
Comment (XmyriyCat) [inline DomainModels/Customer.cs]: Remove all nullables like a '?'

File: Data/Contexts/ProductsDbContext.cs (hunk 1 above)
Comment (KiranitSS) [reply Data/Contexts/ProductsDbContext.cs:26]: Cause Visual Studio say, that here may be null.

Comment (KiranitSS) [reply Data/Contexts/ProductsDbContext.cs:26]: And, I leave the creation of the migration on yourself.

File: DomainModels/Customer.cs (hunk 2 above)
Comment (KiranitSS) [reply DomainModels/Customer.cs]: Okay, np

Comment (KiranitSS) [reply DomainModels/Customer.cs]: Left '?' or "string.Empty"???

Comment (XmyriyCat) [reply DomainModels/Customer.cs]: Remove all usings char '?'

**Review by XmyriyCat — APPROVED** [review]

**Pull request merged**